pkg/errors/errors.go

```diff
@@ -233,7 +233,27 @@ func NewUserNotAllowedError(image string, onbuild bool) error {
 	return Error{
 		Message:    msg,
 		ErrorCode:  UserNotAllowedError,
-		Suggestion: fmt.Sprintf("modify image %q to use a numeric user within the allowed range or build without the --allowed-uids flag", image),
+		Suggestion: fmt.Sprintf("modify image %q to use a numeric user within the allowed range, or build without the --allowed-uids flag", image),
+	}
+}
+
+// NewAssembleUserNotAllowedError returns a new error that indicates that the build
+// could not run because the build or image uses an assemble user outside of the range
+// of allowed users.
+func NewAssembleUserNotAllowedError(image string, assembleUserOverride string) error {
+	var msg, suggestion string
+	if len(assembleUserOverride) > 0 {
+		msg = fmt.Sprintf("assemble user %q must be numeric and within the range of allowed users", assembleUserOverride)
+		suggestion = "build without the --allowed-uids flag or the --assemble-user flag"
+	} else {
+		assembleLabel := "io.openshift.s2i.assemble-user"
+		msg = fmt.Sprintf("image %q includes the %q label whose value is not within the allowed range", image, assembleLabel)
+		suggestion = fmt.Sprintf("modify the %q label in image %q to use a numeric user within the allowed range, or build without the --allowed-uids flag", assembleLabel, image)
```

Comment:
same issue w/ these two messages w/ respect to proposing flags. (assuming i'm right that this text gets exposed as is in openshift)